pkg/service/service.go

```diff
@@ -1,13 +1,125 @@
 package service
 
 import (
+	"context"
+
 	runapi "github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/run"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/config"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/run/v1"
 )
 
-// Client is the connection to update and obtain data about a single service.
+// Client is the connection to update and obtain data about a single service. It
+// contains a client to the Cloud Run API, information about the service and the
+// service object.
 type Client struct {
 	RunClient   runapi.Client
+	Service     *run.Service
 	Project     string
 	ServiceName string
 	Region      string
 }
+
+// NewClient initializes a service client.
+func NewClient(runclient runapi.Client, svc *run.Service, project, serviceName, region string) *Client {
+	return &Client{
+		RunClient:   runclient,
+		Service:     svc,
+		Project:     project,
+		ServiceName: serviceName,
+		Region:      region,
+	}
+}
+
+// Filter retrieves all the services that match the filter in the configuration.
+func Filter(logger *logrus.Logger, cfg *config.Config) []*Client {
+	clientsCh := make(chan []*Client)
+	numberOfFilterByRegionCalls := 0
```

Comment:
`filterByRegionCall` identifier also doesn't make to me, isn't the task here simply:

1. Identify regions (not this method)
2. Choose which regions to use if a region list is specified (not this method)
3. Query all services in all regions "in parallel" + wait
4. As part of (3), query services in 1 region, with label filter.

so I'm not sure which one of 1,2,3,4 I'm reading right now. It seems it's sort of all. :)